src/test/unit/regions/SPRegionTest.cpp

```diff
@@ -303,105 +303,87 @@ namespace testing
 
 TEST(SPRegionTest, testSerialization)
 {
-	  // use default parameters the first time
-	  Network* net1 = new Network();
-	  Network* net2 = nullptr;
-	  Network* net3 = nullptr;
+  // NOTE: this test does end-to-end serialize and deserialize with the following modules:
+  //   Network, Region, Array, ScalerSensor, SPRegion, SpatialPooler, Connections, Random, Links
+  //
+	  // use default parameters
+	  Network net1;
+	  Network net2;
+	  Network net3;
+
+	  VERBOSE << "Setup first network and save it" << std::endl;
+    std::shared_ptr<Region> n1region1 = net1.addRegion("region1", "ScalarSensor", "{n: 100,w: 10,minValue: 1,maxValue: 10}");
+    std::shared_ptr<Region> n1region2 = net1.addRegion("region2", "SPRegion", "{columnCount: 20}");
+    net1.link("region1", "region2", "", "", "encoded", "bottomUpIn");
+    net1.initialize();
+
+    n1region1->setParameterReal64("sensedValue", 5.5);
+		net1.run(1);
+
+    // take a snapshot of everything in SPRegion at this point
+    std::map<std::string, std::string> parameterMap;
+    EXPECT_TRUE(captureParameters(n1region2, parameterMap)) << "Capturing parameters before save.";
+
+    // TODO: JSON serialization does not work.
+    //    returns 3 (not really a crash)
```

Comment:
still valid? (crash)

Reply:
> returns 3 (not really a crash)

Oh, forgot to take out that comment.  That was caused by the stream exceptions. :)

Reply:
Hmmm, went looking for it and that comment was removed.